From d96c777ce8f6a8e95fa3d0fd64d1c0e42bf03adb Mon Sep 17 00:00:00 2001
From: Panu Matilainen <pmatilai@redhat.com>
Date: Fri, 10 May 2024 14:44:04 +0300
Subject: [PATCH] Simplify doBuildDir() and use appropriate macros as well

Realize that rpmExpand() can be passed everything that rstrscat(), only
it expands too. Silly me.
---
 build/build.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/build/build.c b/build/build.c
index 84c88849f7..fd0e4be3f0 100644
--- a/build/build.c
+++ b/build/build.c
@@ -321,12 +321,10 @@ static int doCheckBuildRequires(rpmts ts, rpmSpec spec, int test)
 
 static rpmRC doBuildDir(rpmSpec spec, int test, StringBuf *sbp)
 {
-    char *fix = rpmExpand("%{_fixperms}", NULL);
-    char *doDir = rstrscat(NULL,
-			   "test -d '", spec->buildDir, "' && ",
-			   fix, " '", spec->buildDir, "'\n",
-			   "rm -rf '", spec->buildDir, "'\n",
-			   "mkdir -p '", spec->buildDir, "'\n",
+    char *doDir = rpmExpand("test -d '", spec->buildDir, "' && ",
+			   "%{_fixperms} '", spec->buildDir, "'\n",
+			   "%{__rm} -rf '", spec->buildDir, "'\n",
+			   "%{__mkdir_p} '", spec->buildDir, "'\n",
 			   NULL);
 
     rpmRC rc = doScript(spec, RPMBUILD_MKBUILDDIR, "%mkbuilddir",
@@ -337,7 +335,6 @@ static rpmRC doBuildDir(rpmSpec spec, int test, StringBuf *sbp)
 		spec->buildDir, strerror(errno));
     }
     free(doDir);
-    free(fix);
     return rc;
 }
 
